ios/Exponent/Versioned/Modules/Api/EXContacts.m

```diff
@@ -54,13 +67,54 @@ @implementation EXContacts
 
     NSMutableDictionary *contact = [NSMutableDictionary dictionary];
     contact[@"id"] = person.identifier;
+    contact[@"contactType"] = person.contactType == CNContactTypePerson ? @"person" : @"organization";
+    contact[@"name"] = [self _assembleDisplayNameForContact:person];
     contact[@"firstName"] = person.givenName;
     contact[@"lastName"] = person.familyName;
+    contact[@"previousLastName"] = person.previousFamilyName;
     contact[@"middleName"] = person.middleName;
-    contact[@"name"] = [self _assembleDisplayNameForContact:person];
+    contact[@"nickname"] = person.nickname;
     contact[@"company"] = person.organizationName;
     contact[@"jobTitle"] = person.jobTitle;
+    contact[@"department"] = person.departmentName;
+    contact[@"imageAvailable"] = [NSNumber numberWithBool:person.imageDataAvailable];
 
+    if ([keysToFetch containsObject:CNContactNamePrefixKey]) {
+      contact[@"namePrefix"] = person.namePrefix;
+    }
+    if ([keysToFetch containsObject:CNContactNameSuffixKey]) {
+      contact[@"nameSuffix"] = person.nameSuffix;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticGivenNameKey]) {
+      contact[@"phoneticFirstName"] = person.phoneticGivenName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticMiddleNameKey]) {
+      contact[@"phoneticMiddleName"] = person.phoneticMiddleName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticFamilyNameKey]) {
+      contact[@"phoneticLastName"] = person.phoneticFamilyName;
+    }
+    if ([keysToFetch containsObject:CNContactNoteKey]) {
+      contact[@"note"] = person.note;
+    }
+    if ([keysToFetch containsObject:CNContactImageDataKey]) {
+      contact[@"image"] = @{ @"uri": [NSString stringWithFormat:@"%@%@",
```

Comment:
The size of `imageDate` is the actual image size on disk so I think the only limit is the device's disk space. Thumbnails have very various sizes - I haven't noticed any defined relationship between those sizes.
We discussed it with Krzysztof and as long as there is no official support for blobs and no ability to get an iOS image URI we can't do much. One of the solutions could be creating some custom Image component that gets an URI, parses it and fetches the desired resource.